libs/vkd3d/swapchain.c

```diff
@@ -428,6 +458,13 @@ static ULONG STDMETHODCALLTYPE dxgi_vk_swap_chain_Release(IDXGIVkSwapChain *ifac
 
     if (!refcount)
     {
+        if (chain->queue->device->vk_info.NV_low_latency2)
+        {
+            pthread_mutex_lock(&chain->queue->device->mutex);
```

Comment:
As mentioned earlier, we'll need more fine grained locking, and d3d12_device_add/remove swapchain should take care of this locking, not the swapchain here.

Also, with the proposed reference count increment, this is unsafe, since we're unregistering *after* decrement. This sequence of events is now possible:

- Decrement ref count
- Submission thread queries low latency swapchain, increments refcount
- This thread destroys swapchain
- Submission thread goes boom

Reply:
Adding private refcounts to swapchain should be able to solve this I think.

Reply:
Just make sure that dxgi_vk_swap_chain_drain_queue ref count is reduced to 0 when public ref-count hits 0. If that is called in the submission thread, we get a deadlock.

Reply:
I went ahead and added the private ref count to dxgi_vk_swap_chain and that does seem to make the synchronization easier on the submit thread. Thanks for the suggestion!